File: Ejercicio 11/Ejercicio 11.js
class HashSortSimplificado {
    
    static main() {
        const valores = this.generarValoresAleatorios(15, 100);
        
        console.log("Valores originales:");
        console.log(this.arrayToString(valores));
        const valoresOrdenados = this.hashSortSimplificado(valores);
        
        console.log("\nValores ordenados:");
        console.log(this.arrayToString(valoresOrdenados));
    }
    
    static arrayToString(arr) {
        return "[" + arr.join(", ") + "]";
    }

    static generarValoresAleatorios(cantidad, max) {
        const valores = new Array(cantidad);
        
        for (let i = 0; i < cantidad; i++) {
            valores[i] = Math.floor(Math.random() * max);
        }
        
        return valores;
    }

    static hashSortSimplificado(valores) {
        if (valores.length === 0) {
            return [];
        }
        
        const max = this.encontrarMaximo(valores);
        
        const tablaHash = new Array(max + 1).fill(false);
        
        for (const valor of valores) {
            if (valor >= 0 && valor < tablaHash.length) {
                tablaHash[valor] = true;
            }
        }
        
        const ordenados = [];
        for (let i = 0; i < tablaHash.length; i++) {
            if (tablaHash[i]) {
                ordenados.push(i);
            }
        }
        
        return ordenados;
    }
    
    static encontrarMaximo(valores) {
        let max = valores[0];
        for (let i = 1; i < valores.length; i++) {
            if (valores[i] > max) {
                max = valores[i];
            }
        }
        return max;
    }
    
    static hashSortConTreeSet(valores) {
        const set = new Set();
        
        for (const valor of valores) {
            set.add(valor);
        }
        
        const resultado = Array.from(set);
        resultado.sort((a, b) => a - b);
        
        return resultado;
    }
}

// Ejecutar el programa
HashSortSimplificado.main();

File: Ejercicio 11/Ejercicio 11.cs
using System;
using System.Collections.Generic;

public class HashSortSimplificado
{
    public static void Main(string[] args)
    {
        int[] valores = GenerarValoresAleatorios(15, 100);
        
        Console.WriteLine("Valores originales:");
        Console.WriteLine(ArrayToString(valores));
        int[] valoresOrdenados = HashSortSimplificadoMethod(valores);
        
        Console.WriteLine("\nValores ordenados:");
        Console.WriteLine(ArrayToString(valoresOrdenados));
    }
    
    public static string ArrayToString(int[] arr)
    {
        return "[" + string.Join(", ", arr) + "]";
    }

    public static int[] GenerarValoresAleatorios(int cantidad, int max)
    {
        Random random = new Random();
        int[] valores = new int[cantidad];
        
        for (int i = 0; i < cantidad; i++)
        {
            valores[i] = random.Next(max);
        }
        
        return valores;
    }

    public static int[] HashSortSimplificadoMethod(int[] valores)
    {
        if (valores.Length == 0)
        {
            return new int[0];
        }
        
        int max = EncontrarMaximo(valores);
        
        bool[] tablaHash = new bool[max + 1];
        
        foreach (int valor in valores)
        {
            if (valor >= 0 && valor < tablaHash.Length)
            {
                tablaHash[valor] = true;
            }
        }
        
        List<int> ordenados = new List<int>();
        for (int i = 0; i < tablaHash.Length; i++)
        {
            if (tablaHash[i])
            {
                ordenados.Add(i);
            }
        }
        
        return ordenados.ToArray();
    }
    
    public static int EncontrarMaximo(int[] valores)
    {
        int max = valores[0];
        for (int i = 1; i < valores.Length; i++)
        {
            if (valores[i] > max)
            {
                max = valores[i];
            }
        }
        return max;
    }
    
    public static int[] HashSortConTreeSet(int[] valores)
    {
        SortedSet<int> set = new SortedSet<int>();
        
        foreach (int valor in valores)
        {
            set.Add(valor);
        }
        
        int[] resultado = new int[set.Count];
        int index = 0;
        foreach (int valor in set)
        {
            resultado[index++] = valor;
        }
        
        return resultado;
    }
}